### ISSUE TITLE
console: figure out parent/child spans should be displayed

### ISSUE THREAD

**Issue by davidbarsky** [tokio-rs/console#11]
While #9 is more focused on the data collection side of things, this issue is concerned with how parent/child relationships should be rendered in the console. Possible approaches:
- @hawkw suggests that something like htop's tree mode would be useful.
- being able to dynamically highlight an individual task and view its tree would be also helpful, as @tobz suggested.

**Comment by hawkw** [comment]
Some inspiration:
* [`htop`](https://htop.dev/) tree mode:
  ![Screenshot_20210505_092235](https://user-images.githubusercontent.com/2796466/117175357-99f19700-ad83-11eb-89ca-693d30cc6c43.png)
* [`pstree(1)`](https://man7.org/linux/man-pages/man1/pstree.1.html):
  ![Screenshot_20210505_092421](https://user-images.githubusercontent.com/2796466/117175423-afff5780-ad83-11eb-9593-b068b39a361d.png)
* [`tracing-tree`](https://lib.rs/crates/tracing-tree):
  ![Screenshot_20210505_092634](https://user-images.githubusercontent.com/2796466/117175725-ffde1e80-ad83-11eb-9860-2ca3bb1552b2.png)
* [`reqray`](https://lib.rs/crates/reqray): 
![Screenshot_20210505_092858](https://user-images.githubusercontent.com/2796466/117176057-54819980-ad84-11eb-91ff-58aed3ba56c6.png)
* [Jaeger](https://www.jaegertracing.io/) tree + timeline view:
  ![image](https://user-images.githubusercontent.com/2796466/117176610-e7bacf00-ad84-11eb-80b5-50db8247a71a.png)

I think a tree-mode would probably end up being a different view mode from the `top(1)`-style task list view we have currently. It would also be nice to have a way to highlight one task in the list view and see just _its_ parent context, which we could add to the current view pretty easily, I think. Might be good to start with that before writing a tree-mode?

**Comment by carllerche** [comment]
A timeline view would definitely be awesome, but maybe for the MVP we start by showing the "current state" as a backtrace type view and go from there?

**Comment by hawkw** [comment]
> A timeline view would definitely be awesome, but maybe for the MVP we start by showing the "current state" as a backtrace type view and go from there?

yup, that's what i'm thinking as well. now that #13 gave us the ability to highlight individual tasks in the top tasks view, we have a nice way to select a task and expand it to show the full context. 

tree and timeline views would be really nice to have in a future iteration!
